File: 1413_minimum_value_to_get_positive_step/min_start_value.py
from typing import List


def minStartValue(nums: List[int]) -> int:
    """
    Time: O(n)
    Space: O(1)
    """
    res = curr = 0
    for num in nums:
        curr += num
        res = min(res, curr)
    return -res + 1

File: 1413_minimum_value_to_get_positive_step/min_start_value.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int minStartValue(int[] nums) {
        int min = 0, curr = 0;
        for (int i: nums) {
            curr += i;
            min = Math.min(min, curr);
        }
        return -min + 1;
    }
}